dipy/denoise/localpca.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np
+import scipy as sp
+import scipy.linalg as sla
+import scipy.special as sps
+from scipy.interpolate import interp1d
+from warnings import warn
+
+from dipy.core.ndindex import ndindex
+
+# Try to get the SVD through direct API to lapack:
+try:
+    from scipy.linalg.lapack import sgesvd as svd
+    svd_args = [1, 0]
+# If you have an older version of scipy, we fall back
+# on the standard scipy SVD API:
+except ImportError:
+    from scipy.linalg import svd
+    svd_args = [False]
+
+
+def inv_eta(phi):
+    """
+    The inverse of eta, which corrects for Rician bias at the end of the
+    denoising process (see eqs. 4,5,6 in [Manjon13]_).
+
+    Parameters
+    ----------
+    phi : float
+       signal to noise ratio
+
+    Returns
+    -------
+    inv_eta : float
+        The inverse of the eta function
+    """
+    phi_sq = phi ** 2
+    inv_eta1 = np.log(np.sqrt(np.pi/2)) - phi_sq / 2
+    inv_eta2 = 2 * np.log((((1 + (phi_sq / 2)) * sps.iv(0, phi_sq/4)) +
+                          ((phi_sq / 2) * sps.iv(1, (phi_sq/4)))))
+    return np.exp(inv_eta1 + inv_eta2)
+
+
+def get_eta(start=0, stop=37.4, step=0.01):
+    """
+    Get's an eta function: an interpolated lookup table between SNR
+    and bias corrected.
+    """
+    x = np.arange(start, stop, step)
+    y = inv_eta(x)
+    return interp1d(y, x)
+
+
+def eta(phi, f=None):
+    """
+    A lookup table for correction of Rician bias after denoising (see eqs.
+    4,5,6 in [Manjon13]_).
+
+    Parameters
+    ----------
+    phi : float
+    """
+    if phi < 1.25:
+        return 0
+    else:
+        return f(phi)
+
+
+def localpca(arr, sigma, patch_radius=2, tau_factor=2.3, correct_bias=False):
```

Comment:
correct bias should be turned to correct_rician_bias, to go hand in hand with the rest of the text as that's what it fixes. It should also be true (as the original also uses it, and it would make little sense to have it true for the noise estimation but not the signal part). And it's missing a small explanation of what it does in the docstring.